pkg/filesystem/config.go

```diff
@@ -28,18 +29,18 @@ func WithNydusImageBinaryPath(p string) NewFSOpt {
 
 func WithManager(pm *manager.Manager) NewFSOpt {
 	return func(fs *Filesystem) error {
-		if pm == nil {
-			return errors.New("process manager cannot be nil")
+		if pm != nil {
+			switch pm.FsDriver {
+			case config.FsDriverBlockdev:
+				fs.blockdevManager = pm
```

Comment:
The manager manages all the nydusd daemons which serve nydus image, but looks like tarfs does not need any nydusd daemon. Why there is a blockdev manager which never abord a nydusd daemon.

Reply:
Though tarfs doesn't use daemon, it does use Manager to support:
1) cache space
2) recovery
3) metrics

Reply:
I think we already have a dedicated cache manager which can work alone without a daemon manager.
As tarfs is an erofs mount, nydus-snapshotter does not have to recover any nydus daemon.
Tarfs is running as an Erofs mount without a nydusd daemon, I think there is no metric exported.